config_verifier/src/index.ts

```diff
@@ -0,0 +1,520 @@
+            var targetPeerInformation = {}; 
+            if(target['type'] == 'evm'){
+                targetPeerInformation = await target['manager'].getPeer(src['chainId']);
+            }else if(target['type'] == 'solana'){
+                try{
+                    var targetPeerInformationTmp = await target['manager'].program.account.nttManagerPeer.fetch(target['manager'].peerAccountAddress(src['chainId']));
+                
+                    targetPeerInformation['peerAddress'] = "0x" + bufferToHex(targetPeerInformationTmp['address'])
+                    targetPeerInformation['tokenDecimals'] = targetPeerInformationTmp['tokenDecimals'];
+                }
+                catch(e){ // Don't see peer on Solana
+                    targetPeerInformation["peerAddress"] = "0x000000000000000000000000000000000000000000000" // Can't find the addresss, since it doesn't exist                    
+                    targetPeerInformation['tokenDecimals'] = BigNumber.from(0);
+                }
+            }else{
+                console.log("Invalid chain type...");
+            }
+
+            if(BigNumber.from(targetPeerInformation['peerAddress']).eq(0)){
+                console.log(`ERROR: Peer on chain ${src_id} is not on chain ${target_id}`)
+                continue;
+            }
+
+            // Ensure that the peers match up between the target and destination
+            if(!BigNumber.from(targetPeerInformation['peerAddress']).eq(src['managerAddress'])){
+                console.log(`ERROR: Peers don't match chain id for manager sender chain ${target_id} and receiving chain ${src_id}`);
+            }
+
+            // Check that the decimals for the configuration match up with the other chains decimals
+            if(targetPeerInformation['tokenDecimals'] != src['tokenDecimals']){
+                console.log(`ERROR: Peers don't match decimals for sender chain ${target_id} with decimals ${targetPeerInformation['tokenDecimals']} and receiving chain ${src_id} with decimals ${src['tokenDecimals']}`);
+            }  
+
+            var inboundRateLimitParams; 
+            if(target['type'] == 'evm'){ 
+                // Inbound rate limit is sane
+                inboundRateLimitParams = await target['manager'].getInboundLimitParams(src['chainId'] );
+
+            }else if(target['type'] == 'solana'){
+                inboundRateLimitParams = (await target['manager'].program.account.inboxRateLimit.fetch(target['manager'].inboxRateLimitAccountAddress(src['chainId'])))['rateLimit'];
+            }
+            
+            // Rate limit is set to 0. Indicates that it's disabled.
+            if(inboundRateLimitParams['limit'].eq(0)){
+                console.log(`WARNING: Inbound rate limit disabled for sender chain ${target_id} and target chain ${src_id}`); 
+                continue;
+            }
+
+            // Don't feel this is necessary or proper. Depends on the use case.
```

Comment:
IMO this is a nice check, especially since you flag it as a warning rather than an error.